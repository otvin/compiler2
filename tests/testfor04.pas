program testfor04(output);
{overwriting the initial and final values will not change the number of iterations.  Example: Cooper p.29}


var a,b,Counter:integer;

begin
	writeln('should print ''Le plus ca change...'' 3 times');
	writeln('');

	a := 1;
	b := 3; 
	for Counter := a to b
		do begin
			writeln('Le plus ca change...');
			a := -2000;  {These assignments have no}
			B := 2001	 {effect on the for statement.}
		end
end.
