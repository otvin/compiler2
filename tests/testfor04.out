should print 'Le plus ca change...' 3 times

Le plus ca change...
Le plus ca change...
Le plus ca change...
